src/renderer/components/swap/Swap.tsx

```diff
@@ -734,6 +749,57 @@ export const Swap = ({
     )
   }, [canSwitchAssets, setPendingSwitchAssets, assetsToSwap, onChangePath])
 
+  const [isApprovedRD] = useObservableState<RD.RemoteData<ApiError, boolean>>(
+    () =>
+      FP.pipe(
+        sourcePoolRouter,
+        O.map((router) => {
+          if (isERC20Asset(sourceAssetProp)) {
+            return isApprovedERC20Token$({
+              spender: router,
+              sender: getEthTokenAddress(ERC20Assets.filter((asset) => eqAsset.equals(asset, sourceAssetProp))[0])
```

Comment:
`getEthTokenAddress(sourceAssetProp)` ?

Reply:
The reason for it is because the sourceAssetProp address starts with 0X, not 0x and `getEthTokenAddress` doesn't handle it. But once https://github.com/thorchain/asgardex-electron/pull/1074 is merged, then it's fine to use it.

Reply:
Ok, changes of #1074  are available in this PR now (see 2d13611)